batch7crud-roll7/src/main/java/com/lftechnology/batch7crud/controller/StudentListController.java

```diff
@@ -0,0 +1,187 @@
+        }
+        catch(NumberFormatException e){
+            showErrorPage(request, response);
+        }
+        int totalStudents = stdservice.fetchTotal();
+
+        if(page !=1 && page > Math.ceil(totalStudents/(float)pageSize)){
+            showErrorPage(request,response);
+        }
+        request.setAttribute("students", stdservice.fetch(page,pageSize));
+        request.setAttribute("pageSize",pageSize);
+        request.setAttribute("totalStudents",totalStudents);
+        request.setAttribute("pageNum",page);
+
+        request.getServletContext().getRequestDispatcher("/WEB-INF/views/students.jsp").forward(request, response);
+
+    }
+
+    private void create(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        request.getServletContext().getRequestDispatcher("/WEB-INF/views/newEntry.jsp").forward(request, response);
+    }
+
+    private void createProcess(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException, DataException {
+        String name = request.getParameter("name");
+        String address = request.getParameter("address");
+        String roll = request.getParameter("roll");
+        try{
+
+            int rollNum = Integer.parseInt(roll);
+            Student s = new Student();
+            s.setName(name);
+            s.setAddress(address);
+            s.setRoll(rollNum);
+
+            StudentServices stdServices = new StudentServices();
+            stdServices.addNew(s);
+            response.sendRedirect("/Students");
+        }catch(NumberFormatException e){
+            request.setAttribute("error","invalid roll");
+            request.getServletContext().getRequestDispatcher("/WEB-INF/views/newEntry.jsp").forward(request, response);
+        }
+
+
+    }
+
+    private void edit(HttpServletRequest request, HttpServletResponse response, int id)
+            throws ServletException, IOException, DataException {
+        StudentServices stdservice = new StudentServices();
+
+        request.setAttribute("student", stdservice.fetchById(id));
+        request.getRequestDispatcher("/WEB-INF/views/editEntry.jsp").forward(request, response);
+
+    }
+
+    private void editProcess(HttpServletRequest request, HttpServletResponse response, int id)
+            throws ServletException, IOException, DataException {
+        String name = request.getParameter("name");
+        String address = request.getParameter("address");
+        int roll = Integer.parseInt(request.getParameter("roll"));
+        Student s = new Student();
+        s.setName(name);
+        s.setAddress(address);
+        s.setRoll(roll);
+        s.setId(id);
+
+        StudentServices stdServices = new StudentServices();
+        stdServices.update(s, id);
+        response.sendRedirect("/Students");
+
+    }
+
+    private void deleteProcess(HttpServletRequest request, HttpServletResponse response, int roll)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused method parameter(s) "request". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1172)